connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -892,6 +893,9 @@ message MapPartitions {
 
   // (Optional) Whether to use barrier mode execution or not.
   optional bool is_barrier = 3;
+
+  // (Optional) ResourceProfile id used for the stage level scheduling.
+  optional int32 profile_id = 4;
```

Comment:
Why do we need an extra RPC for that? Can't you attach the resource profile directly to this call?

Reply:
Hi @grundprinzip, The user still needs to know the exact ResourceProfile id, if we attach resource profile in the call, seems we can't get id in this call.